src/fmemopen.c

```diff
@@ -126,28 +127,22 @@ FILE *fmemopen(void *buf, size_t len, const char *type)
 	FILE *fp;
 	char tp[MAX_PATH - 13];
 	char fn[MAX_PATH + 1];
-	HANDLE h;
 
-	if (!GetTempPath(sizeof(tp), tp))
+	if (!GetTempPathA(sizeof(tp), tp))
 		return NULL;
 
-	if (!GetTempFileName(tp, "confuse", 0, fn))
+	if (!GetTempFileNameA(tp, "confuse", 0, fn))
 		return NULL;
 
-	h = CreateFile(fn, GENERIC_READ | GENERIC_WRITE, 0, NULL,
-		       CREATE_ALWAYS, FILE_FLAG_DELETE_ON_CLOSE, NULL);
-	if (INVALID_HANDLE_VALUE == h)
+	fd = _open(fn,
```

Comment:
Reading the docs for _open [1], it states that the optional pmode argument is no longer optional with this combination of flags (specifically _O_CREAT and the docs are even more specific when that is combined with either of _O_SHORT_LIVED or _O_TEMPORARY).

Also from reading various docs, CreateFile creates an handle which is not inheritable (by default), but _open does. _O_NOINHERIT is desired, methinks.

[1] https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen?view=vs-2019

Reply:
@peda-r: So the resulting call should look more like this?

```c
fd = _open(fn, _O_CREAT | _O_RDWR | _O_SHORT_LIVED | _O_TEMPORARY | _O_BINARY,
               _S_IREAD | _S_IWRITE | _O_NOINHERIT, _S_IREAD | _S_IWRITE);
```